core/domain/state_domain.py

```diff
@@ -1673,20 +1673,18 @@ def update_interaction_answer_groups(self, answer_groups_list):
                 self.interaction.answer_groups)]
         # TODO(yanamal): Do additional calculations here to get the
         # parameter changes, if necessary.
-        for answer_group_dict in answer_groups_list:
-            rule_specs_list = answer_group_dict['rule_specs']
+        for answer_group in answer_groups_list:
+            rule_specs_list = answer_group.rule_specs
             if not isinstance(rule_specs_list, list):
                 raise Exception(
                     'Expected answer group rule specs to be a list, '
                     'received %s' % rule_specs_list)
-
-            answer_group = AnswerGroup(
-                Outcome.from_dict(answer_group_dict['outcome']), [],
-                answer_group_dict['training_data'],
-                answer_group_dict['tagged_skill_misconception_id'])
-            for rule_dict in rule_specs_list:
-                rule_spec = RuleSpec.from_dict(rule_dict)
-
+            normalized_answer_group = AnswerGroup(
```

Comment:
Can't you just do `answer_group.rule_specs = []` and use that itself? Since we already have it in object form.

Reply:
Done, sorry for the delay. I got some heavier workloads recently.